# Question 4
-----

In [1]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
            
        return x

In [2]:
z = normal(0, 1)
loc_1 = z.locate(0.98)
loc_2 = z.locate(0.01)
print(loc_1)
print(loc_2)

2.053759999947009
-2.326330000064751


$$58.37 - 2.05\sigma = \mu$$

$$2.32\sigma + 40.85 = \mu$$

$$58.37 - 2.05\sigma = 2.32\sigma + 40.85 $$

$$4.37\sigma = 17.52$$

$$\sigma = 4$$

$$\mu = 58.37 - (2.05 * 4)$$

$$\mu = 50.17$$